### ISSUE TITLE
Support for refresh_tokens

### ISSUE THREAD

**Issue by bumi** [getAlby/lndhub.go#83]
currently we do not support the refresh tokens.
If a user provides a valid refresh token we should return an access token.

Actually I also do not really understand the current use of the refresh vs. access token. They are kinda the same and do not really provide more security as far as I understand.

But we should be backwards compatible here to the old lndhub.
So maybe for now we simply allow the /auth call to be called with login/password or the refresh_token and if valid we return new access and refresh tokens.

**Comment by skosito** [comment]
Hey @bumi @kiwiidb , let's quickly discuss this before implementing, just to make sure we are all on same page.

To be compatible with lndhub API we need refresh tokens. This is imo simple, but good golang reference implementation we can use https://dev.to/abrichak/jwt-and-go-how-to-integrate-them-with-security-requirements-eh5.

We would need to store tokens somehow, it looks like redis is good option, it would allow us to not store this short-lived data in main db, we will access it often, it is faster, and there is convenient way to clear record from cache after X minutes which is automatically logging out users. Lndhub is using redis for this as well, but from what I saw they are using redis for everything.

Of course drawback of this is adding additional storage, but maybe it is worth it in this case.

Additionally, this implementation is also checking if user exists in db in ValidateToken method, but I think that is not needed for first version.

Please check and let me know your thoughts on this. If you approve I will start implementing this.

**Comment by bumi** [comment]
hmm. I did not read that article, yet. But wanted to share my quick thoughts: 
Initially we went with JWT because it is stateless and we do not need to store token data. I think this is a great advantage because we do not need to deal with storage and invalidation of the tokens at all. At the same time we do not have the chance to invalidate specific tokens, but this hasn't been a requirement so far. 

One of the key design goals is also to have as little run time dependencies as possible. So I am very hesitant to add something like Redis and would even say this is a no-go right now. For the use case that you describe Redis is for sure great, but it just adds complexity on the deployment and maintenance.

I actually also think that the current use of login+password and access_token/refresh_token is problematic. Clients simply store the login/password and use that to get an access token to do API calls. 
BlueWallet implements getting a new access_token with a refresh_token but if the refresh_token is expired or if the refresh fails it just uses the login/password to get new tokens. (https://github.com/BlueWallet/BlueWallet/blob/f105acb0b67a11079e756d586d4712ed485262f0/class/wallets/lightning-custodian-wallet.js#L280-L299)
Then why not just always use login/password?

This is not an oAuth flow where apps would only get an access/refresh token. So the login/password are the weakest part...Even if we improve access/refresh token usage somehow there is still the problem that a login/password even exists and are stored by clients. 

So with these thoughts I actually don't think it justifies much additional complexity or code currently. 

I would simply add the option that the `/auth` endpoint can be called with the refresh_token (additionally to login+password) and that's it. 

(btw. BlueWallet assumes the access_token is valid for 2 days, and the refresh_token is valid for 7 days: https://github.com/BlueWallet/BlueWallet/blob/f105acb0b67a11079e756d586d4712ed485262f0/class/wallets/lightning-custodian-wallet.js#L83-L89)
